cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -369,15 +367,14 @@ impl YaccParser {
         // We pass around an index into the *bytes* of self.src. We guarantee that at all times
         // this points to the beginning of a UTF-8 character (since multibyte characters exist, not
         // every byte within the string is also a valid character).
-        let section_required = matches!(self.yacc_kind_resolver, YaccKindResolver::NoDefault);
-        let header_parser = GrmtoolsSectionParser::new(&self.src, section_required);
+        let header_parser = GrmtoolsSectionParser::new(&self.src, false);
         let result = match header_parser.parse() {
-            Ok((mut header, i)) => {
-                let result = self.update_yacckind(&mut header, i);
-                self.header = header;
-                result
+            Ok((parsed_header, i)) => {
+                self.header.merge_from(parsed_header).unwrap();
```

Comment:
This is `unwrap()` is just a temporary hack that was done just to get things working, 
These errors should somehow be converted into a grammar error.

Reply:
I fixed that by trying to map this error to one of the `Duplicate` errors `DuplicateGrmtoolsSectionEntry`.

[06390fa](https://github.com/softdevteam/grmtools/pull/547/commits/06390fa0030dda28769140947d431ad97df480be)
This subsequently required me to add a new spans kind for none-or-more spans in [6a7f5c5](https://github.com/softdevteam/grmtools/pull/547/commits/6a7f5c52d73ff978f4479ef0a4d0af31fd8e3ab6)

I'm not certain that reusing `DuplicateGrmtoolsSectionEntry` is the right thing, e.g. usually we see duplicate because of something silly like:

```
%grmtools{yacckind: Grmtools, yacckind: Grmtools}
```

however this is like

```
cp_builder.yacckind(YaccKind::Grmtools)
```

Parsing a source file with:

```
%grmtools{yacckind: Grmtools}
```